ibm/resource_ibm_private_dns_glb_monitor.go

```diff
@@ -0,0 +1,520 @@
+package ibm
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/IBM/dns-svcs-go-sdk/dnssvcsv1"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	pdnsGlbMonitorName          = "name"
+	pdnsGlbMonitorID            = "monitor_id"
+	pdnsGlbMonitorDescription   = "description"
+	pdnsGlbMonitorType          = "type"
+	pdnsGlbMonitorPort          = "port"
+	pdnsGlbMonitorInterval      = "interval"
+	pdnsGlbMonitorRetries       = "retries"
+	pdnsGlbMonitorTimeout       = "timeout"
+	pdnsGlbMonitorMethod        = "method"
+	pdnsGlbMonitorPath          = "path"
+	pdnsGlbMonitorAllowInsecure = "allow_insecure"
+	pdnsGlbMonitorExpectedCodes = "expected_codes"
+	pdnsGlbMonitorExpectedBody  = "expected_body"
+	pdnsGlbMonitorHeaders       = "headers"
+	pdnsGlbMonitorHeadersName   = "name"
+	pdnsGlbMonitorHeadersValue  = "value"
+	pdnsGlbMonitorCreatedOn     = "created_on"
+	pdnsGlbMonitorModifiedOn    = "modified_on"
+)
+
+func resourceIBMPrivateDNSGLBMonitor() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMPrivateDNSGLBMonitorCreate,
+		Read:     resourceIBMPrivateDNSGLBMonitorRead,
+		Update:   resourceIBMPrivateDNSGLBMonitorUpdate,
+		Delete:   resourceIBMPrivateDNSGLBMonitorDelete,
+		Exists:   resourceIBMPrivateDNSGLBMonitorExists,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			pdnsGlbMonitorID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Monitor Id",
+			},
+
+			pdnsInstanceID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Instance Id",
+			},
+
+			pdnsGlbMonitorName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique identifier of a service instance.",
+			},
+
+			pdnsGlbMonitorDescription: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Descriptive text of the load balancer monitor",
+			},
+
+			pdnsGlbMonitorType: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "HTTP",
+				ValidateFunc: validateAllowedStringValue([]string{"HTTP", "HTTPS", "TCP"}),
```

Comment:
Use Latest way of validation.. Please Refer to CIS Resources

Reply:
Handled